File: 1256-rank-transform-of-an-array/1256-rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        
        map<int,int> mp;


        priority_queue<int, vector<int>, greater<int>> pq;
        for(auto it : arr)pq.push(it);

        int i =1;
        while(!pq.empty()){
            int ele = pq.top(); pq.pop();
            if(mp.contains(ele)){
                mp[ele] = min(mp[ele], i);
                // i++;
            }else {
                mp[ele] = i;
                i++;
            }
        }
        for(int i=0; i<arr.size(); i++){
            arr[i] = mp[arr[i]];
        }

        
        return arr;
    }
};

File: 1256-rank-transform-of-an-array/1256-rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        
        unique_list = sorted(list(set(arr)))
        mp = {}

        for i in range(len(unique_list)):
            mp[unique_list[i]] = i+1

        for i in range(len(arr)):
            arr[i] = mp[arr[i]]

        return arr